File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
	public int ExpectedMinutesInOven() => 40;
	public int RemainingMinutesInOven(int minutesInOven) => ExpectedMinutesInOven() - minutesInOven;
	public int PreparationTimeInMinutes(int numberLayers) => numberLayers * 2;
	public int ElapsedTimeInMinutes(int numberLayers, int minutesInOven) =>
		PreparationTimeInMinutes(numberLayers) + RemainingMinutesInOven(minutesInOven);
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40
let remainingMinutesInOven minutes = expectedMinutesInOven - minutes
let preparationTimeInMinutes numberOfLayers = numberOfLayers * 2
let elapsedTimeInMinutes numberOfLayers minutes = preparationTimeInMinutes(numberOfLayers) + minutes

